src/main/java/com/dougnoel/sentinel/files/CsvFile.java

```diff
@@ -0,0 +1,334 @@
+package com.dougnoel.sentinel.files;
+
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import com.dougnoel.sentinel.exceptions.FileException;
+import com.dougnoel.sentinel.strings.SentinelStringUtils;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class CsvFile extends TestFile{
+    private static final Logger log = LogManager.getLogger(CsvFile.class.getName()); // Create a logger.
+
+    private final CSVFormat csvFormat;
+    private final int numHeaderRows;
+
+    /**
+     * Create default CSV file from most recently-downloaded path.
+     * @throws FileNotFoundException in the case that a file does not exist at that path.
+     */
+    public CsvFile() throws FileNotFoundException {
+        this(1);
+    }
+
+    /**
+     * Create default CSV file from specified path.
+     * @param pathToFile Path path to the CSV file.
+     * @throws FileNotFoundException in the case that a file does not exist at that path.
+     */
+    public CsvFile(Path pathToFile) throws FileNotFoundException {
+        super(pathToFile);
+        csvFormat = CSVFormat.DEFAULT;
+        numHeaderRows = 1;
+    }
+
+    /**
+     * Create a CSV file with the given file format from the CSV file at the specified path.
+     * @param pathToFile Path path to the CSV file.
+     * @param format CSVFormat the format of the CSV file. See CSVFormat documentation for more info.
+     * @throws FileNotFoundException in the case that a file does not exist at that path.
+     */
+    public CsvFile(Path pathToFile, CSVFormat format) throws FileNotFoundException {
+        super(pathToFile);
+        csvFormat = format;
+        numHeaderRows = 1;
+    }
+
+    /**
+     * Create a CSV file configured with the given number of header rows from the CSV file at the most recently downloaded path.
+     * @param numberOfHeaderRows int number of header rows in the CSV file.
+     * @throws FileNotFoundException in the case that a file does not exist at that path.
+     */
+    public CsvFile(int numberOfHeaderRows) throws FileNotFoundException {
+        super();
+        csvFormat = CSVFormat.DEFAULT;
+        numHeaderRows = numberOfHeaderRows;
+    }
```

Comment:
This constructor isn't exercised. I'd like to see it removed if we aren't using it now or have a unit test covering it so someone trying to use it later isn't surprised when the code doesn't work.